### PR TITLE
feat(openapi-tooling): add support for fuzzing with schemathesis

### PR DESCRIPTION
**Pull request by aramissennyeydd**
## Hey, I just made a Pull Request!

This adds fuzzing capabilities for plugins documented with OpenAPI using [Schemathesis](https://schemathesis.readthedocs.io/en/stable/index.html) 🎉 !

search and todo were able to integrate with no issues, but the catalog has some API weirdnesses. Basically, defining `fields` for `/by-refs` or `/entities` causes the return value to be missing "required" fields. Would be curious to hear thoughts on if this is just something we ignore. It would be a breaking change on the client, but it's already codified behavior in the API.

Following this work, I'd like to make this into a GH workflow for use with Uffizzi deployments.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/repo-tools/src/commands/index.ts [hunk 1]
```diff
@@ -59,6 +59,25 @@ function registerPackageCommand(program: Command) {
         import('./package/schema/openapi/generate').then(m => m.command),
       ),
     );
+
+  openApiCommand
+    .command('fuzz')
+    .description(
+      'Fuzz an OpenAPI schema by generating random data and sending it to the server.',
+    )
+    .option('--count <count>', 'Number of requests to send')
```
[the remark this answers is not in the document]
Comment (aramissennyeydd) [reply packages/repo-tools/src/commands/index.ts:68]: Good call, renaming to `limit` might make this a little more clear?

[the remark this answers is not in the document]
Comment (Stranger6667) [reply packages/repo-tools/src/commands/index.ts:68]: Yep! :)

**Comment by aramissennyeydd** [comment]
not stale :)

**Review by mikeyhc — APPROVED** [review]

File: packages/repo-tools/src/commands/package/schema/openapi/fuzz.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  if (opts.limit) {
+    args.push('--hypothesis-max-examples', opts.limit);
+  }
+  args.push('--workers', opts.workers);
+
+  if (opts.useGuest) {
+    // TODO: @sennyeya This should leverage the `guest-provider` if available.
+    args.push('--header', `Authorization: Basic guest`);
```
Comment (Rugvip) [inline packages/repo-tools/src/commands/package/schema/openapi/fuzz.ts:62]: Thinking we might have add a few hardcoded tokens that work in development, similar to the mock service

**Review by Rugvip — APPROVED** [review]
Sweet! 👏 

Nice to see it already found some stuff, gonna :shipit: 🎉

**Comment by Stranger6667** [comment]
Happy to see this merged! If you’ll ever need any support from the Schemathesis side, feel free to tag me.

**Pull request merged**